stonesoup/models/measurement/tests/test_combined.py

```diff
@@ -72,10 +72,10 @@ def test_rvs(model):
 
     rvs_state = model.rvs(10)
     assert rvs_state.shape == (4, 10)
-    assert all(isinstance(state, Bearing) for state in rvs_state[0])
-    assert all(not isinstance(state, Bearing) for state in rvs_state[1])
-    assert all(isinstance(state, Bearing) for state in rvs_state[2])
-    assert all(not isinstance(state, Bearing) for state in rvs_state[3])
+    assert all(isinstance(state[0], Bearing) for state in rvs_state[0])
+    assert all(not isinstance(state[0], Bearing) for state in rvs_state[1])
+    assert all(isinstance(state[0], Bearing) for state in rvs_state[2])
+    assert all(not isinstance(state[0], Bearing) for state in rvs_state[3])
```

Comment:
I was a little confused by this change, but now looking more closely, as I can see it's an issue caused by ending up with `StateVectors` type when slicing in certain ways. The new iteration method has highlight this bug here.

I'll push commit here with fix for that, which probably will also resolve some other issues.